sockeye/encoder.py

```diff
@@ -206,8 +241,12 @@ def encode(self, data: mx.sym.Symbol, data_length: mx.sym.Symbol, seq_len: int)
         :param seq_len: Maximum sequence length.
         :return: Encoded input data.
         """
+        current_data_length = data_length
+        current_seq_len = seq_len
         for encoder in self.encoders:
-            data = encoder.encode(data, data_length, seq_len)
+            data = encoder.encode(data, current_data_length, current_seq_len)
+            current_data_length = encoder.get_encoded_data_length(current_data_length)
+            current_seq_len = encoder.get_encoded_seq_len(current_seq_len)
```

Comment:
Encoder sequences propagate length information (both max sequence length and the actual lengths of each sequence).